subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -157,9 +158,21 @@ static struct net_buf *cdc_acm_buf_alloc(struct usbd_class_data *const c_data,
  * common (UDC) buffer, as this results in a smaller footprint.
  */
 static struct net_buf *cdc_acm_buf_alloc(struct usbd_class_data *const c_data,
-					 const uint8_t ep)
+					 const uint8_t ep, size_t size_hint)
 {
-	return usbd_ep_buf_alloc(c_data, ep, USBD_MAX_BULK_MPS);
+	/* The USB stack handles packetizing to the endpoint size, therefore
+	 * allocating buffers largers than the endpoint size reduces the number
+	 * of round trips through this class driver when the TX FIFO size is
+	 * larger than the endpoint, improving throughput.
+	 *
+	 * Limit the maximum allocation size to 1/4 of the total pool size
+	 * to prevent starving other USB users. Ensure the maximum allocation
+	 * never drops below USBD_MAX_BULK_MPS.
+	 */
```

Comment:
>	/* The USB stack handles packetizing to the endpoint size, therefore

"USB stack" does nothing like that. That is the UDC driver (and sometimes controller) who handles a transfer, transaction by transaction. Please change this comment to be just:

```c
	/*
	 * Limit the maximum allocation size to 1/4 of the UDC pool size
	 * to prevent starving other pool users. Ensure the maximum allocation
	 * never drops below USBD_MAX_BULK_MPS.
	 */
```
Please fix your commit message as well.

Reply:
Your higher reasoning is redundant. There is not need to explain how the UDC driver works. Remove it as suggested.

Reply:
Why is it redundant? It's explaining why the function is allocating buffer sizes other than MPS. That seems like an important point for a function called `cdc_acm_buf_alloc`.